project/HaKkaz/README.md

```diff
@@ -0,0 +1,67 @@
+NSD Term Project Proposal
+---
+
+## Basic Information
+GitHub repository: [Point in Polygon](https://github.com/HaKkaz/Point-in-Polygon)
+
+The Point in Polygon problem is a fundamental problem in computational geometry that involves determining whether a given point lies inside, outside, or on the boundary of a polygon.
+
+## Problem to Solve
+The basic point-in-polygon problem is easy to solve when the polygon is contain $N$ edges and there is only one point in each query, using the Ray Casting algorithm, with a time complexity of $O(N)$.
+
+However, this library addresses a more advanced problem where each query may contain $M$ points, and it solves it with a time complexity of $O((N+M) \log_{2}{(N+M)})$.
+
+## Prospective Users
+For users who need to develop geospatial software, many software services now require the use of mapping functionality, especially applications involving location or geographic information. This is particularly true when dividing areas, as seen in services like Uber, Airbnb, Pokemon GO, etc.
+
+## System Architeure
+- Using ``C++`` to implement the algorithm.
+- Using ``pybind11`` to make API in python.
+
+## API Description
+
+- `C++` Types
+```cpp
+template<typename T>
+class Point {
+    T x, y;
+}
+
+template<typename T>
+class Segment {
+    Point<T> a,b;
+    int id;
+}
+
+template<typename T>
+class Polygon {
+    private:
+        std::vector<Point<T>> points;
```

Comment:
You assume a polygon is a node list.  How do you deal with polygons with holes?

Reply:
Currently, I use array of polygons to maintain a polygon with hole. The Ray Casting algorithm can handle the polygon with hole by checking the number of segement be intersected to the ray is even or odd.

Reply:
I am not seeing how an array of polygons can be used for representing holes.  Please keep in mind it may be an issue.

It will be good to make it explicit for how you will handle it in your proposal and/or code and presentation.  Not supporting holes is also a way to handle it.